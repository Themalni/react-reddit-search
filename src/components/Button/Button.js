import styled from 'styled-components'

const Button = styled.button`
  display: block;
  width: 100%;
  padding: 1em 2.5em;
  border: none;
  background: linear-gradient(to bottom, ${(props) => props.theme.colors.orange}, ${(props) => props.theme.colors.yellowOrange});
  color: ${(props) => props.theme.colors.white};
  border-radius: 4px;
  cursor: pointer;
  font-size: 1em;
  position: relative;
  z-index: 1;

  &:before {
    content: "Search";
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    border-radius: inherit;
    background: linear-gradient(to top, ${(props) => props.theme.colors.orange}, ${(props) => props.theme.colors.yellowOrange});
    opacity: 0;
    z-index: 2;
    line-height: 3.2;
    transition: opacity .4s linear;
  }

  &:hover {
    &:before {
      transition: opacity .4s linear;
      opacity: 1;
    }
  }

  ${({ theme }) => theme.media.tablet} {
    display: inline-block;
    width: 100%;
  }
`

const WideButton = styled(Button)`
  display: block;
  margin: 0 auto;
  width: 100%;
  padding: 0.7em 0;
`
const LinkButton = styled.a`
  display: block;
  margin: 0 auto;
  width: 60%;
  text-decoration: none;
  text-align: center;
  padding: 0.5em 0;
  background: linear-gradient(to bottom, ${(props) => props.theme.colors.orange}, ${(props) => props.theme.colors.yellowOrange});
  color: ${(props) => props.theme.colors.white};
  border-radius: 4px;
  cursor: pointer;
  font-size: 1em;
  position: relative;

  &:before {
    content: "Read more";
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    border-radius: inherit;
    background: linear-gradient(to top, ${(props) => props.theme.colors.orange}, ${(props) => props.theme.colors.yellowOrange});
    opacity: 0;
    z-index: 2;
    line-height: 2.45;
    transition: opacity .4s linear;
  }

  &:hover {
    &:before {
      transition: opacity .4s linear;
      opacity: 1;
    }
  }
`

export { Button, WideButton, LinkButton }