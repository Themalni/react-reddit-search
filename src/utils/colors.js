export const colors = {
  white: '#fff',
  graphite: '#666',
  grey: '#8a96a3',
  yellowOrange: '#ff9933',
  orange: '#ed5f55',
  orangeDark: '#d8574e',
  blue: '#3a81b7',
  blueLight: '#64a0ce'
};

// d02216