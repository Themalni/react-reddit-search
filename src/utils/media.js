export const media = {
  phone: '@media (min-width: 320px)',
  tablet: '@media (min-width: 728px)',
  laptop: '@media (min-width: 1024px)',
  desktop: '@media (min-width: 1280px)'
}