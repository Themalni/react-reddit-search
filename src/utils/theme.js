import { colors } from './colors';

export const theme = {
  colors,
  font: {
    thin: 300,
    regular: 400,
    bold: 700,
  },
  media: {
    phone: '@media (min-width: 320px)',
    tablet: '@media (min-width: 728px)',
    laptop: '@media (min-width: 1024px)',
    desktop: '@media (min-width: 1280px)'
  }
}