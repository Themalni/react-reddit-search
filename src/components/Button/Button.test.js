import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components'
import { theme } from '../../utils/theme';
import { Button, WideButton, LinkButton } from './Button';
import { ThemeProvider } from 'styled-components';

describe('Container', () => {
  it("renders correctly", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Button />
      </ThemeProvider>
    )
    expect(container.firstChild).toMatchSnapshot();
  })
})

describe('LowerContainer', () => {
  it("renders correctly", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <WideButton />
      </ThemeProvider>
    )
    expect(container.firstChild).toMatchSnapshot();
  })
})

describe('MasonryContainer', () => {
  it("renders correctly", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <LinkButton />
      </ThemeProvider>
    )
    expect(container.firstChild).toMatchSnapshot();
  })
})