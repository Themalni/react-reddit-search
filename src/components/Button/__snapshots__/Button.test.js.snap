// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders correctly 1`] = `
.c0 {
  display: block;
  width: 100%;
  padding: 1em 2.5em;
  border: none;
  background: linear-gradient(to bottom,#ed5f55,#ff9933);
  color: #fff;
  border-radius: 4px;
  cursor: pointer;
  font-size: 1em;
  position: relative;
  z-index: 1;
}

.c0:before {
  content: "Search";
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  border-radius: inherit;
  background: linear-gradient(to top,#ed5f55,#ff9933);
  opacity: 0;
  z-index: 2;
  line-height: 3.2;
  -webkit-transition: opacity .4s linear;
  transition: opacity .4s linear;
}

.c0:hover:before {
  -webkit-transition: opacity .4s linear;
  transition: opacity .4s linear;
  opacity: 1;
}

@media (min-width:728px) {
  .c0 {
    display: inline-block;
    width: 100%;
  }
}

<button
  class="c0"
/>
`;

exports[`LowerContainer renders correctly 1`] = `
.c0 {
  display: block;
  width: 100%;
  padding: 1em 2.5em;
  border: none;
  background: linear-gradient(to bottom,#ed5f55,#ff9933);
  color: #fff;
  border-radius: 4px;
  cursor: pointer;
  font-size: 1em;
  position: relative;
  z-index: 1;
  display: block;
  margin: 0 auto;
  width: 100%;
  padding: 0.7em 0;
}

.c0:before {
  content: "Search";
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  border-radius: inherit;
  background: linear-gradient(to top,#ed5f55,#ff9933);
  opacity: 0;
  z-index: 2;
  line-height: 3.2;
  -webkit-transition: opacity .4s linear;
  transition: opacity .4s linear;
}

.c0:hover:before {
  -webkit-transition: opacity .4s linear;
  transition: opacity .4s linear;
  opacity: 1;
}

@media (min-width:728px) {
  .c0 {
    display: inline-block;
    width: 100%;
  }
}

<button
  class="c0"
/>
`;

exports[`MasonryContainer renders correctly 1`] = `
.c0 {
  display: block;
  margin: 0 auto;
  width: 60%;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  padding: 0.5em 0;
  background: linear-gradient(to bottom,#ed5f55,#ff9933);
  color: #fff;
  border-radius: 4px;
  cursor: pointer;
  font-size: 1em;
  position: relative;
}

.c0:before {
  content: "Read more";
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  border-radius: inherit;
  background: linear-gradient(to top,#ed5f55,#ff9933);
  opacity: 0;
  z-index: 2;
  line-height: 2.45;
  -webkit-transition: opacity .4s linear;
  transition: opacity .4s linear;
}

.c0:hover:before {
  -webkit-transition: opacity .4s linear;
  transition: opacity .4s linear;
  opacity: 1;
}

<a
  class="c0"
/>
`;
